### PR TITLE
Fix a fatal client issue, a light level issue, & make use of ITickable in OpenLightTE for setting states.

### PR DESCRIPTION
**Pull request by Korynkai**
Eh, commit message is mostly self-explanitory:

* Fixed an issue where the client would fatally error due to the wrong TE finding its way into ClientProxy.BlockColorHandler.colorMultiplier(). (Also fixes a fatal issue currently present in OpenLightTE.getBrightness())

* Fixed an issue where vanilla light was not actually being emitted (may have only occurred in testing).

* Implemented ITickable from OpenLightTE so we can just use an internal flag to trigger updates. (note: update() runs in the server thread anyway, so we can do this instead of scheduling a task)

* A little cleanup (some unnecessary translations, checks, etc...)

Also IntelliJ decided to reformat some code now that I'm using it -- this is based on our code-formatting guidelines, so any code coming from myself or QMX should be within such guidelines.

### CODE REVIEW

**Comment by ben-mkiv** [comment]
thanks, but please dont mind if some of the reformating might be undone at some point, as its unreadable on 1280x1024 screen without lots of scrolling

then about the tile using the update() method, im not a big fan. Did you test if this works as expected when calling setColor() and setBrightness() in a Lua script without delay? Because its not really threadsafe to set the tick flag in oc thread and to reset it in world thread. this would require something like a mutex which forces one thread to wait before it can set it to true again, so i would rather suggest to keep it event based instead of letting the world thread poll for changes

**Comment by Korynkai** [comment]
Yeah the reformatting kinda crept in, I could revert those formatting changes if you'd like...

However, I wasn't quite sure about using update() either due to thread safety at first, but it behaves exactly as expected. (__NOTE__: Not the preferred method; see "__tl;dr__") (This change really wasn't really planned to be this extensive, was only supposed to patch a couple bugs, but this was determined to fix one of the bugs, so...)

As it is, `updateOnTick()` only flips the private `updateOnTick` flag from false to true, and it is not flipped back except by the main thread on update.  In this usage, it simply acts as a signal to the main thread to process the update so long as the `updateOnTick` field is never changed outside of `updateOnTick()` and `update()`.  I've used this pattern in the past in similar cases with no adverse affects.

Perhaps an option would be to synchronize `updateOnTick()` or use a true `java.util.concurrent.Semaphore` instead. However, the only time a bug would be encountered using the code in this PR would be with rapid cycling at a rate approaching the tick rate, and at that it would jump to the latest values, not the earlier, which is usually what we want if we're running at that rate -- lets assume the server locked up for a second: less memory to allocate and process because instead of inserting a new `Runnable` object on the scheduled task list, it now has to do a check on the tile entity as a reference, which would occur somewhere around the same time, when the server is iterating through `ITickable.update()` calls rather than when it is iterating through `Runnable` objects in the scheduled task list -- but never repeatedly on the same tick.  If there are no changes this tick and the `updateOnTick` flag is false, processing continues unaffected.  With a scheduled task, we force that update each time there is a change, forcing it to use each set of values sequentially (in some lockup situations, even forcing it to update _before_ the last tick has completed, causing two updates in one tick, which may be undesirable, especially given that this would most likely occur within the lockup situation anyway and create more for the server to process).  With `update()` we flag down that a change has occurred, and, if the server doesn't get to it in time for the next change to occur, it uses the latest values. (Furthermore, this eliminates some scoping issues, such as unnecessary casting...)

**tl;dr: _However_** it has just come to my attention there was a solution within OpenComputers this entire time.  It seems rather than using `@Callback(direct=true)`, we should be using `@Callback(direct=false)` as this runs these updates within the server thread anyway, eliminating this unnecessary abstraction.  Will test to ensure this works within expected parameters and push when I am satisfied with the results.

**Comment by Korynkai** [comment]
Excuse the comment edit; stumbling on a nuance in the annotation that I should've thought of to begin with apparently...  I'll be pushing a new commit as soon as possible; there's a few things I still feel the need to test.

**Comment by ben-mkiv** [comment]
afaik the direct annotation just lets the computer sleep for 1/20 second

**Comment by fnuecke** [comment]
Just to confirm what @Korynkai said, yes, `direct=true` means it called directly from the worker thread, `direct=false` means the call is evaluated in Minecraft's server thread. The former is a lot faster (as it doesn't need to sync) and typically used for stuff like component-local data mutation (e.g. the graphics cards). `direct=false` should be used for anything that interacts with the world or other systems that aren't thread safe.

**Comment by ben-mkiv** [comment]
Thanks, a new version (0.1.0.7) is on curseForge which has all the PRs in it
